app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java

```diff
@@ -138,6 +155,231 @@ public static void playOnBackgroundPlayer(final Context context, final PlayQueue
         startService(context, getPlayerIntent(context, BackgroundPlayer.class, queue, resumePlayback));
     }
 
+    public static void downloadPlaylist(final PlaylistFragment context, final PlayQueue queue) {
+        List<PlayQueueItem> events = queue.getStreams();
+        Iterator<PlayQueueItem> eventsIterator = events.listIterator();
+        if (eventsIterator.hasNext()) {
+            startDownloadPlaylist(context, eventsIterator, null);
+        }
+    }
+
+    private static void startDownloadPlaylist(final PlaylistFragment activity,
+                                              final Iterator<PlayQueueItem> itemIterator,
+                                              DownloadSetting downloadSetting) {
+        if (downloadSetting != null) {
+            Completable.create(emitter -> {
```

Comment:
The create method should not be used. Also this isn't the ideal place to subscribe.